### DISCUSSION TITLE
Open telemetry metrics in Weblogic

### DISCUSSION THREAD

**Discussion by ssarraju** [oracle/weblogic-monitoring-exporter#207] [Q&A]
Team, 

We want to use open telemetry annotations and instrumentation to capture application metrics and have them pulled into prometheus. How can we achieve this in an app running on Weblogic?

Thanks
Sravan

**Comment by rjeberhard** [comment]
Hi Sravan,

This is not something that we've looked at. Much like this monitoring exporter project, I suspect that open telemetry data can be scraped from a metrics endpoint that is either deployed on WebLogic as a Java EE web application (WAR) or running next to WebLogic, such as in a Kubernetes sidecar container.

We've done something similar with the Prometheus client servlet here: https://github.com/prometheus/client_java/blob/main/simpleclient_servlet/src/main/java/io/prometheus/client/exporter/MetricsServlet.java

I see that open telemetry supports a Java agent, as here: https://opentelemetry.io/docs/instrumentation/java/automatic/

Likely, your application would have the necessary annotations, you would run the Java agent, and then you would scrape from Prometheus using a MetricsServlet, as above. There may need to be connective code as in this sample: https://github.com/open-telemetry/opentelemetry-java-docs/blob/main/prometheus/src/main/java/io/opentelemetry/example/prometheus/PrometheusExample.java.

Are you using WebLogic on-premise, on compute, or in Kubernetes?

Looping in @mriccell, @robertpatrick

**Reply by ssarraju** [reply]
We are using Weblogic on on-premise for majority and a handful on compute.
